packages/application-shell/src/configure-store.ts

```diff
@@ -51,10 +54,23 @@ const mergeObjectValues = <T>(object: { [key: string]: T }) =>
 const patchedGetCorrelationId = () =>
   getCorrelationId({ userId: selectUserId() });
 
+const getAdditionalHeaders = (): Headers | undefined => {
+  const sessionToken = window.sessionStorage.getItem(
+    STORAGE_KEYS.SESSION_TOKEN
+  );
+  return omitEmpty<Headers>({
+    [SUPPORTED_HEADERS.AUTHORIZATION]: sessionToken
+      ? `Bearer ${sessionToken}`
+      : undefined,
+    [SUPPORTED_HEADERS.X_APPLICATION_ID]: window.app.applicationId,
+    [SUPPORTED_HEADERS.X_TEAM_ID]: selectTeamIdFromLocalStorage(),
+  });
+};
+
 const sdkMiddleware = createSdkMiddleware({
   getCorrelationId: patchedGetCorrelationId,
   getProjectKey: selectProjectKeyFromUrl,
-  getTeamId: selectTeamIdFromLocalStorage,
+  getAdditionalHeaders,
```

Comment:
We're passing now a different function as it better encapsulates the required headers.